libs/ml/include/ml/core/graph.hpp

```diff
@@ -240,7 +243,11 @@ void Graph<TensorType>::Step(DataType learning_rate)
   {
     auto trainable_ptr = std::dynamic_pointer_cast<ops::Trainable<TensorType>>(t->GetOp());
     trainable_ptr->Step(learning_rate);
-    ResetGraphCache(t, false);
+  }
+
```

Comment:
TODO(issue number) - we should not reset graph cache for ALL nodes, but just trainable nodes in the relevant part of the graph....?